### ISSUE TITLE
[BUG] Multiple winget processes 

### ISSUE THREAD

**Issue by fernandog** [marticliment/WingetUI#2334]
### Please confirm these before moving forward

- [X] I have searched for my issue and have not found a work-in-progress/duplicate/resolved issue.
- [X] I have tested that this issue has not been fixed in the latest [(beta or stable) release](https://github.com/marticliment/WingetUI/releases/).
- [X] I have checked the [FAQ](https://github.com/marticliment/WingetUI#frequently-asked-questions) section for solutions
- [X] This issue is about a bug (if it is not, please use the [correct template](https://github.com/marticliment/WingetUI/issues/new/choose))

### Describe your issue

looks like is locking the winget.exe for another application

![image](https://github.com/marticliment/WingetUI/assets/2620870/82a5c5b8-c1e0-42bb-a4dd-68ada6bd7c60)

### Steps to reproduce the issue

Load wingetui and load the other application. Doesn't happen always, I'm still trying to understand 

### WingetUI Log

```text
Version 3.1.0-alpha1
```

### Package Managers Logs

```text
.
```

**Comment by marticliment** [comment]
You will need to give more details on the issue, I don't understand what the issue is

**Comment by fernandog** [comment]
I'm trying to figure it out. is it normal to have all those winget process running at the same time?

**Comment by fernandog** [comment]
![image](https://github.com/marticliment/WingetUI/assets/2620870/474a7022-4967-4d06-85b2-c12692d88413)

Happened again, but nothing wrong with the logs:


```
> [22 jun. 2024 09:52:43] 
>                            __  __      _ ______     __  __  ______
>                           / / / /___  (_) ____/__  / /_/ / / /  _/
>                          / / / / __ \/ / / __/ _ \/ __/ / / // /  
>                         / /_/ / / / / / /_/ /  __/ /_/ /_/ // /   
>                         \____/_/ /_/_/\____/\___/\__/\____/___/   
>                             Welcome to UniGetUI Version 3.1.0-alpha1
> [22 jun. 2024 09:52:43]
> [22 jun. 2024 09:52:43] Version Code:  3,0991
> [22 jun. 2024 09:52:44] Loaded language locale: en
> [22 jun. 2024 09:52:44] Using bundled GSudo at C:\Program Files\WingetUI\Assets\Utilities\gsudo.exe
> [22 jun. 2024 09:52:45] Api auth token: apj7joc9zw418pwnrxdvqr35zletewb5uqsv5zqnp7pdn9ishtyv1npjc73hid2n
> [22 jun. 2024 09:52:46] Api running on http://localhost:7058
> [22 jun. 2024 09:52:46] Lang files were updated successfully from GitHub
> [22 jun. 2024 09:52:46] UniGetUI is up to date
> [22 jun. 2024 09:52:46] Command npm was not found on the system
> [22 jun. 2024 09:52:46] ▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄
>                         █▀▀▀▀▀▀▀▀▀▀▀▀▀ MANAGER LOADED ▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀
>                         █ Name: Npm
>                         █ Enabled: True
>                         █ Found: False
>                         █ THE MANAGER WAS NOT FOUND. PERHAPS IT IS NOT 
>                         █ INSTALLED OR IT HAS BEEN MISCONFIGURED 
>                         ▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀
> [22 jun. 2024 09:52:46] Downloaded new icons and screenshots successfully!
> [22 jun. 2024 09:52:46] ▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄
>                         █▀▀▀▀▀▀▀▀▀▀▀▀▀ MANAGER LOADED ▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀
>                         █ Name: .NET Tool
>                         █ Enabled: True
>                         █ Found: True
>                         █ Fancye exe name: dotnet tool
>                         █ Executable path: C:\Program Files\dotnet\dotnet.exe
>                         █ Call arguments: tool
>                         █ Version: 
>                         █   
>                         ▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀
> [22 jun. 2024 09:52:47] Command scoop was not found on the system
> [22 jun. 2024 09:52:47] ▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄
>                         █▀▀▀▀▀▀▀▀▀▀▀▀▀ MANAGER LOADED ▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀
>                         █ Name: Scoop
>                         █ Enabled: False
>                         █ THE MANAGER IS DISABLED
>                         ▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀
> [22 jun. 2024 09:52:47] ▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄
>                         █▀▀▀▀▀▀▀▀▀▀▀▀▀ MANAGER LOADED ▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀
>                         █ Name: Winget
>                         █ Enabled: True
>                         █ Found: True
>                         █ Fancye exe name: winget.exe
>                         █ Executable path: C:\Users\Fernando\AppData\Local\Microsoft\WindowsApps\winget.exe
>                         █ Call arguments: 
>                         █ Version: 
>                         █   Naive WinGet CLI Version: v1.7.11261
>                         █   Using Native WinGet helper (COM Api)
>                         ▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀
> [22 jun. 2024 09:52:47] ▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄
>                         █▀▀▀▀▀▀▀▀▀▀▀▀▀ MANAGER LOADED ▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀
>                         █ Name: Chocolatey
>                         █ Enabled: False
>                         █ THE MANAGER IS DISABLED
>                         ▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀
> [22 jun. 2024 09:52:47] ▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄
>                         █▀▀▀▀▀▀▀▀▀▀▀▀▀ MANAGER LOADED ▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀
>                         █ Name: Pip
>                         █ Enabled: True
>                         █ Found: True
>                         █ Fancye exe name: pip
>                         █ Executable path: C:\Users\Fernando\AppData\Local\Programs\Python\Python312\python.exe
>                         █ Call arguments:  -m pip
>                         █ Version: 
>                         █   pip 24.1 from C:\Users\Fernando\AppData\Local\Programs\Python\Python312\Lib\site-packages\pip (python 3.12)
>                         ▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀
> [22 jun. 2024 09:52:49] ▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄▄
>                         █▀▀▀▀▀▀▀▀▀▀▀▀▀ MANAGER LOADED ▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀
>                         █ Name: PowerShell
>                         █ Enabled: True
>                         █ Found: True
>                         █ Fancye exe name: powershell.exe
>                         █ Executable path: C:\WINDOWS\system32\windowspowershell\v1.0\powershell.exe
>                         █ Call arguments:  -NoProfile -Command
>                         █ Version: 
>                         █   Name                           Value                                                                                   
>                         █   ----                           -----                                                                                   
>                         █   PSVersion                      5.1.22621.3646                                                                          
>                         █   PSEdition                      Desktop                                                                                 
>                         █   PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}                                                                 
>                         █   BuildVersion                   10.0.22621.3646                                                                         
>                         █   CLRVersion                     4.0.30319.42000                                                                         
>                         █   WSManStackVersion              3.0                                                                                     
>                         █   PSRemotingProtocolVersion      2.3                                                                                     
>                         █   SerializationVersion           1.1.0.1
>                         ▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀▀
> [22 jun. 2024 09:52:49] LoadComponentsAsync finished executing. All managers loaded. Proceeding to interface.
> [22 jun. 2024 09:52:49] Command dotnet-tools-outdated.exe was not found on the system
> [22 jun. 2024 09:52:49] Error finding updates on manager .NET Tool
> [22 jun. 2024 09:52:49] System.InvalidOperationException: Cannot start process because a file name has not been provided.
>                            at System.Diagnostics.Process.Start()
>                            at UniGetUI.PackageEngine.Managers.DotNetManager.DotNet.GetAvailableUpdates_UnSafe() in C:\SomePrograms\WingetUI-Store\src\UniGetUI.PackageEngine.Managers.Dotnet\DotNet.cs:line 65
>                            at UniGetUI.PackageEngine.ManagerClasses.Manager.PackageManager.GetAvailableUpdates() in C:\SomePrograms\WingetUI-Store\src\UniGetUI.PackageEngine.PackageManagerClasses\Manager\PackageManager.cs:line 177
> [22 jun. 2024 09:52:49] Found 0 installed packages from .NET Tool
> [22 jun. 2024 09:52:51] Found 133 installed packages from Pip
> [22 jun. 2024 09:52:53] Found 3 installed packages from PowerShell
> [22 jun. 2024 09:52:57] Found 0 available updates from PowerShell
> [22 jun. 2024 09:52:58] Found 225 installed packages from Winget
> [22 jun. 2024 09:52:58] Found 2 available updates from Winget
> [22 jun. 2024 09:53:11] Found 0 available updates from Pip
> [22 jun. 2024 09:54:11] Command dotnet-tools-outdated.exe was not found on the system
> [22 jun. 2024 09:54:11] Error finding updates on manager .NET Tool
> [22 jun. 2024 09:54:11] System.InvalidOperationException: Cannot start process because a file name has not been provided.
>                            at System.Diagnostics.Process.Start()
>                            at UniGetUI.PackageEngine.Managers.DotNetManager.DotNet.GetAvailableUpdates_UnSafe() in C:\SomePrograms\WingetUI-Store\src\UniGetUI.PackageEngine.Managers.Dotnet\DotNet.cs:line 65
>                            at UniGetUI.PackageEngine.ManagerClasses.Manager.PackageManager.GetAvailableUpdates() in C:\SomePrograms\WingetUI-Store\src\UniGetUI.PackageEngine.PackageManagerClasses\Manager\PackageManager.cs:line 177
> [22 jun. 2024 09:54:15] Found 2 available updates from Winget
> [22 jun. 2024 09:54:16] Found 0 available updates from PowerShell
> [22 jun. 2024 09:54:17] Found 0 available updates from Pip
```

**Comment by fernandog** [comment]
@marticliment any clue what's going on?

**Comment by marticliment** [comment]
I suspect the issue you are experiencing is related to a different issue, fixed on a commit between the latest alhpa and today.

I'll post soon the next prerelease so you can test if it gets fixed or not

**Comment by marticliment** [comment]
This issue has been fixed by [`UniGetUI 3.1.0 beta 0`](https://github.com/marticliment/WingetUI/releases/tag/3.1.0-beta0).
